velox/dwio/dwrf/writer/ColumnWriter.cpp

```diff
@@ -2036,9 +2118,17 @@ std::unique_ptr<BaseColumnWriter> BaseColumnWriter::create(
     case TypeKind::INTEGER:
       return std::make_unique<IntegerColumnWriter<int32_t>>(
           context, type, sequence, onRecordPosition);
-    case TypeKind::BIGINT:
+    case TypeKind::BIGINT: {
+      if (type.type()->isDecimal()) {
+        return std::make_unique<DecimalColumnWriter>(
+            context, type, sequence, onRecordPosition);
+      }
```

Comment:
Noob question, I saw reader having a different filter, is this expected?
https://github.com/facebookincubator/velox/blob/bfc199fc65ced0a4bf7644ea5314e6ec5fa69eea/velox/dwio/dwrf/reader/ReaderBase.cpp#L288

Reply:
Thanks for noticing. I assume we need to fix the reader base so as to get the correct fileType. In the ColumnWriterTest, the reader is created with 'ColumnReader::build' using the test data type so it does not trigger above mismatch.
https://github.com/facebookincubator/velox/blob/d1bf9da3d4d36685569578eacea771088c1c9330/velox/dwio/dwrf/reader/ColumnReader.cpp#L2459-L2465

Reply:
Correction: dwrf does not include a 'decimal' file type, so the decimal column reader and writer are both for ORC file format. This PR adds 'format' in 'WriterOptions' to distinguish between DWRF and ORC when writing. Thanks!